### PR TITLE
txnbuild: Updates package with Protocol 14 changes.

### PR DESCRIPTION
**Pull request by Shaptic**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What
Basically, I ran the following commands (simplified)

 - `rsync -c -r -u internal/txnbuild ../../`
 - find & replace "/services/horizon/internal/txnbuild" with "/txnbuild"
 - `rm -rf internal/txnbuild`

All integration & unit tests pass.

### Why
We want to make our Protocol 14 updates visible.

### Known limitations
Because of the `-u` flag on the `rsync` command, I **only** copied new[er] files. This means that if someone updated txnbuild with a change after we updated the _same file_ inside of internal/txnbuild, **it wouldn't get copied.** I don't think this is the case anywhere and tried confirming with `diff`, but a sanity check would be wonderful.

Questions:  
 - Do we need to update the `CHANGELOG.md`? 
 - Or make version bumps anywhere

### CODE REVIEW

**Comment by Shaptic** [comment]
Okay wait this is super incomplete, please don't review yet :sweat_smile:

**Review by abuiles — CHANGES_REQUESTED** [review]
heads up, you are removing a bunch of valid files (they were removed from the internal txnbuild to make static check  happy, but we shouldn't be removing files).

**Comment by 2opremio** [comment]
@Shaptic If you want to be 100% sure, it would worth trying out https://stackoverflow.com/questions/1214906/how-do-i-merge-a-sub-directory-in-git

**Review by tamirms — APPROVED** [review]

**Review by abuiles — APPROVED** [review]
